support-files/mariadb-socket-convert

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+#
+# Used to generate a mariadb.socket file based on the curent mysql/maridb settings
+#
+# This is to assist distro maintainers in migrating to systemd socket activation from
+# a user system with networking settings somewhere in the my.cnf files.
+#
+# Redirect output to user directory like /etc/systemd/system/mariadb.socket.d/migrated-from-my.cnf-settings.conf
+
+
+print_ports()
+{
+if [ -n "${port}" ]; then
+  echo "ListenStream=${bind}${port}"
+fi
+if [ -n "${extraport}" ]; then
+  echo "ListenStream=${bind}${extraport}"
+fi
+}
+
+bindaddress=
+port=
+extraport=
+backlog=
+maxconnections=
+skip_networking=
+socket=
+
+coproc mysqldefaults { /usr/bin/my_print_defaults --mysqld ; }
+while read var; do
+  key=${var%%=*}
+  val=${var#*=}
+  case "${key}" in
+    --bind-address)
+      bindaddress=$val
+      ;;
+    --bind_address)
+      bindaddress=$val
+      ;;
+    --port)
+      port=$val
+      ;;
+    --extra-port)
+      extraport=$val
+      ;;
+    --extra_port)
```

Comment:
works for me. will change
<pre>
$ key=--extra_port
$ case "${key}" in --extra[-_]port) echo yes=$key; ;; esac
yes=--extra_port
$ key=--extra-port
$ case "${key}" in --extra[-_]port) echo yes=$key; ;; esac
yes=--extra-port
</pre>